test/integration/order.test.ts

```diff
@@ -77,4 +81,38 @@ describe('order', () => {
       owner: selfAlias,
     })
   })
+
+  it('creates an order transaction on chain', async () => {
+    demandCreateMock()
+    const { body: { id: orderId } } = await post(app, '/order', { parametersAttachmentId })
+
+    // submit to chain
+    const response = await post(app, `/order/${orderId}/creation`, {})
+    expect(response.status).to.equal(201)
+
+    const { id: transactionId, state } = response.body
+    expect(transactionId).to.match(
+      /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89ABab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$/
+    )
+    expect(state).to.equal(TransactionState.submitted)
+
+    // check local transaction updates
+    const [transaction] = await db.getTransaction(transactionId)
+    expect(transaction.state).to.equal(TransactionState.finalised)
+    expect(transaction).to.contain({
+      state: 'finalised',
+      apiType: 'order',
+      transactionType: 'creation',
+    })
+
+    // check local capacity updates with token id
+    const [order] = await db.getDemand(orderId)
+    expect(order).to.contain({
+      id: orderId,
+      owner: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY',
+      state: 'created',
+      subtype: 'order',
+      parametersAttachmentId,
+    })
+  })
 })
```

Comment:
could do with more tests to check the routes added `GET order/id` `GET order/` + sad path tests

Reply:
Could you please explain what scenarios are not covered?
I just did not cover for get `Validation Error`, also what about that tool? Also sad/happy path sounds weird, we need some scenarios, right?

A rule of thumb for  testing (one of) to test bad scenarios first and then everything else becomes a happy path without extra wrappers, and bad scenarios do trigger errors and we should assert error and also validate that other method have not been called after the error.

Basically, I can walk you through and share some of mine knowledge when it comes to TDD

Reply:
I see more tests have been added since I commented. There's no `200` test for `GET order/id` nor one testing the `400` on order creation `if (order.state !== DemandState.created)`

Reply:
yea, totally forgot that added get controllers, only for demo. This has been updated

Reply:
still not seeing a test for `if (order.state !== DemandState.created)`?

Reply:
must have forgotten this one. Updated. thanks.